### DISCUSSION TITLE
Best practices for storing changing sizes of (text) data on documents, considering RIDs?

### DISCUSSION THREAD

**Discussion by SevorisDoe** [ArcadeData/arcadedb#1660] [General]
I think this is a simple thing really, but I am unclear about the behavior here, given that RIDs cannot change without destroying associations recorded elsewhere: assuming I am storing various bits of text data and other metadata on documents in a bucket, and I rewrite an existing documents so that the data would bump into the start of a subsequent allocated record of the page - what's the behavior here? Does the transaction fail? Is the rest of the data implicitly written to a new RID allocated further down the file? Should one maybe use referenced documents and use larger record spacings per pages to avoid issues in advance?

**Comment by lvca** [comment]
If I understood your question, records larger than a page (or don't fit the space left in the page), are split and allocated in multiple pages with a linked list pointer structure. It's all transparent to the user.

**Reply by SevorisDoe** [reply]
My question was more aimed at what happens if a record, through updates over time, grows to the point of colliding with other records allocated on the same page. But I gather that subsumes this same behavior of using linked lists to chain larger records together from chunks. 

Thanks for quickly coming back to this!

**Reply by lvca** [reply]
Yes, it's always split over multiple pages.
